### PR TITLE
Add "networkx.plugin_info" entry point and update docstring

### PR DESCRIPTION
**Pull request by eriknw**
We use a new entry point so it can be imported really fast by not needing to import the backend.

Companion PR for `nx-cugraph` is here: https://github.com/rapidsai/cugraph/pull/3848

Needs documented, discussed, etc. I don't know how well the updated docstrings look like yet or how well they will render online.

CC @rlratzel @betochimas

### CODE REVIEW

**Comment by MridulS** [comment]
Shouldn't we also install no deps cugraph or graphblas to test out the doc building?

**Comment by MridulS** [comment]
Hmm that looks nice but would sphinx be happy rendering these custom numpy doc strings blocks? Maybe we could add admonitions https://pydata-sphinx-theme.readthedocs.io/en/stable/examples/kitchen-sink/admonitions.html to make them pop on the rendered doc site?

**Comment by eriknw** [comment]
@MridulS, I figured out how to show an extra section in online docs when using numpydoc! I'm curious how palatable you'll find the workaround--I monkeypatch `numpydoc`, but I hope it's not _too_ bad.

I got the idea looking at sympy discussions around this; see the changed linked to here: https://github.com/sympy/sympy/issues/15370#issuecomment-428731466

Let's hope the docs build fine for the latest commit 🤞 

Also, I fixed a few docstring errors that I saw.

I know we still need to update the graphblas backend to be able to show this information. Any other backends ready to be added to the online docs?

Oh, and this currently uses `nx-cugraph` nightly wheels. Should we stick to releases instead?

**Comment by eriknw** [comment]
Check it out:
- https://output.circle-artifacts.com/output/job/a8746d1b-05e9-47d4-84d9-fffccbc9e4ba/artifacts/0/doc/build/html/reference/algorithms/generated/networkx.algorithms.community.louvain.louvain_communities.html#networkx.algorithms.community.louvain.louvain_communities
- https://output.circle-artifacts.com/output/job/a8746d1b-05e9-47d4-84d9-fffccbc9e4ba/artifacts/0/doc/build/html/reference/algorithms/generated/networkx.algorithms.centrality.betweenness_centrality.html#networkx.algorithms.centrality.betweenness_centrality

**Comment by MridulS** [comment]
> I got the idea looking at sympy discussions around this; see the changed linked to here: https://github.com/sympy/sympy/issues/15370#issuecomment-428731466

lol

**Review by jarrodmillman — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -38,6 +38,10 @@ jobs:
             pip install -r requirements/extra.txt
             pip install -r requirements/example.txt
             pip install -r requirements/doc.txt
+            # Install trusted backends, but not their dependencies.
+            # We only need to use the "networkx.plugin_info" entry-point.
+            # This is the nightly wheel for nx-cugraph.
+            pip install nx-cugraph-cu11 --extra-index-url https://rapidsdev:WeArePyPI@pypi.k8s.rapids.ai/simple --no-deps
```
Comment (MridulS) [inline .circleci/config.yml:44]: We should probably add this to pyproject.toml with something like `backend-docs` to have a single source of truth but we can do this when we update this action with graphblas docs.

Thanks @eriknw !

**Pull request merged**